src/App.js

```diff
@@ -30,25 +29,76 @@ const App = () => {
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [currPlayer, setCurrentPlayer] = useState(PLAYER_1);
+  const [currWinner, setWinner] =useState('No one yet')
+
+  const updateSquareData = (updatedSquare) => {
+
+    // const flatarr = squares.flat();
+    const newSquares = squares.map(row =>{
+      for (let j =0; j<row.length; j++){
+        const square = row[j]
+        if (square.id === updatedSquare.id) {
+          square.value = currPlayer;
+        }  
+      }
+      return row;    
+    });
+    setSquares(newSquares);
+    if (currPlayer === PLAYER_1) {
+      setCurrentPlayer(PLAYER_2)
+    } else {
+      setCurrentPlayer(PLAYER_1)
+
+    }
+    let finalWinner = checkForWinner(newSquares)
+    console.log(finalWinner);
+    setWinner(finalWinner);
+    console.log('checkforwinner function passed sucessfully!')
```

Comment:
Remember to clean up debug logging:

```suggestion
    setWinner(finalWinner);
```